### PR TITLE
Add a highly available Citus deployment via Stackgres Operator

### LINKED ISSUE #5233: Citus investigate high availability
### Problem

Mirrornode database is a typical multi-tenant use case in that we have many different entities and thus entity-related tables. The official citus recommendation to achieve high availability for such use case is utilizing postgresql's WAL streaming replication to have at least one standby for each coordinator as well as each worker.

[pg_auto_failover](https://github.com/citusdata/pg_auto_failover), an extension developed by citus, supports citus specific WAL streaming replication to hot standby nodes as well as auto failover.

We need to investigate and understand how it can be integrated with our helm chart deployment.

### Solution

- Follow the official pg_auto_failover guide to create a 3-worker node cluster with hot standbys
- Understand how streaming replication and failover is configured
- Propose the approach to integrate pg_auto_failover with our helm chart deployment 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add the Stackgres cluster configuration to `hedera-mirror` chart
* Add the Stackgres Operator to the `hedera-mirror-common` chart
* Change `/api/v1/contracts/call` to return `Service Unavailable 503` for query timeout
* Fix REST API logging a stacktrace for statement timeouts
* Fix statement timeout not working for Java components
* Fix unnecessary `Processed 0 sidecars` logs in importer

**Related issue(s)**:

Fixes #5233

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: charts/hedera-mirror/templates/secret-passwords.yaml [hunk 1]
```diff
@@ -1,50 +1,149 @@
 {{- $name := "mirror-passwords" -}}  # postgresl-ha doesn't support templated names
+{{- $dbHost := include "hedera-mirror.db" . }}
+{{- $dbHostPrimary := printf "%s-coord-primary" (include "hedera-mirror.stackgres" .) }}
+{{- $dbName := .Values.db.name }}
+{{- $dbSchema := .Values.db.schema }}
+{{- $secret := lookup "v1" "Secret" (include "hedera-mirror.namespace" .) $name | default dict -}}
```
Comment (steven-sheehy) [inline charts/hedera-mirror/templates/secret-passwords.yaml:6]: It's a bit nasty to have all component's db connection info and db scripts in one file, but it's currently necessary due to the lookup function to ensure we atomically generate passwords or lookup existing.

File: charts/hedera-mirror/templates/secret-passwords.yaml [hunk 2]
```diff
@@ -1,50 +1,149 @@
...
-  PGPOOL_POSTGRES_CUSTOM_USERS: "{{ $graphQLUsername }},{{ $grpcUsername }},{{ $importerUsername }},{{ $ownerUsername }},{{ $restUsername }},{{ $rosettaUsername }},{{ $web3Username }}"
+  PGPOOL_POSTGRES_CUSTOM_PASSWORDS: "{{ $graphqlPassword }},{{ $grpcPassword }},{{ $importerPassword }},{{ $ownerPassword }},{{ $restPassword }},{{ $rosettaPassword }},{{ $web3Password }}"
+  PGPOOL_POSTGRES_CUSTOM_USERS: "{{ $graphqlUsername }},{{ $grpcUsername }},{{ $importerUsername }},{{ $ownerUsername }},{{ $restUsername }},{{ $rosettaUsername }},{{ $web3Username }}"
   password: {{ coalesce .Values.postgresql.postgresql.password (get $passwords "password" | default "" | b64dec) (randAlphaNum 40) | quote }}
   repmgr-password: {{ coalesce .Values.postgresql.postgresql.repmgrPassword (get $passwords "repmgr-password" | default "" | b64dec) (randAlphaNum 40) | quote }}
   {{ end -}}
+
+  {{- if .Values.stackgres.enabled }}
+  all-postgres.sql: |
+    create or replace function if_role_not_exists(rolename name, sql text) returns text as
```
Comment (steven-sheehy) [inline charts/hedera-mirror/templates/secret-passwords.yaml:69]: Create user/role does not have a `if not exists` option to ensure idempotency. There's also a race condition where the coordinator node can create the user first and sync it to worker nodes, but does not sync existing roles to new workers added later.

**Review by edwin-greene — DISMISSED** [review]
Looks good!

**Review by jascks — DISMISSED** [review]
LGTM

File: charts/hedera-mirror/templates/stackgres/stackgres-cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    instances: {{ .Values.stackgres.coordinator.instances }}
+    managedSql:
+      scripts:
+        - sgScript: {{ include "hedera-mirror.stackgres" . }}-coordinator
+    pods:
+      persistentVolume: {{ .Values.stackgres.coordinator.persistentVolume | toYaml | nindent 8 }}
+    sgInstanceProfile: {{ include "hedera-mirror.stackgres" . }}-coordinator
+  database: {{ .Values.db.name }}
+  nonProductionOptions:
+    disableClusterPodAntiAffinity: true
```
Comment (jnels124) [inline charts/hedera-mirror/templates/stackgres/stackgres-cluster.yaml:22]: doesn't this allow more than one db server to exist on the same host? Was this intentional to keep?

**Review by jnels124 — APPROVED** [review]

File: charts/hedera-mirror/templates/stackgres/stackgres-cluster.yaml (hunk 3 above)
Comment (steven-sheehy) [reply charts/hedera-mirror/templates/stackgres/stackgres-cluster.yaml:22]: Good catch. I meant to remove this but need to keep it for CI.

Comment (steven-sheehy) [reply charts/hedera-mirror/templates/stackgres/stackgres-cluster.yaml]: Fixed

File: docs/configuration.md [hunk 4]
```diff
@@ -309,6 +309,7 @@ value, it is recommended to only populate overridden properties in the custom `a
 | `hedera.mirror.grpc.db.name`                                | mirror_node      | The name of the database                                                                                                         |
 | `hedera.mirror.grpc.db.password`                            | mirror_grpc_pass | The database password used to connect to the database.                                                                           |
 | `hedera.mirror.grpc.db.port`                                | 5432             | The port used to connect to the database                                                                                         |
+| `hedera.mirror.grpc.db.statementTimeout`                    | 10000            | The maximum amount of time in seconds to wait for a query to finish                                                              |
```
Comment (xin-hedera) [inline docs/configuration.md:312]: in milliseconds?

Comment (steven-sheehy) [reply docs/configuration.md]: Fixed

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**